furryfeet_be/furryfeet/views.py

```diff
@@ -104,3 +104,156 @@ def post(self, request, pk):
 
         serializer = UserSerializer(user)
         return Response(serializer.data, status=status.HTTP_200_OK)
+    
+
+class DogWalkerView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+          raise AuthenticationFailed("Unauthenticated!")
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+         
+        try:
+            dog_walker = DogWalker.objects.get(user=request.user)
+            serializer = DogWalkerSerializer(dog_walker)
+            appointments = dog_walker.appointments.all()
+            feedbacks = dog_walker.feedbacks.all()
+            return Response({'dog_walker': serializer.data, 'appointments': AppointmentSerializer(appointments, many=True).data, 'feedbacks': FeedbackSerializer(feedbacks, many=True).data})
+        except DogWalker.DoesNotExist:
+            return Response(status=404)
+
+    def post(self, request, pk):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+          raise AuthenticationFailed("Unauthenticated!")
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+
+        try:
+            if payload['id'] == pk:
+                user = User.objects.get(pk=pk)
+            else:
+                raise AuthenticationFailed("Unauthenticated!")
+        except User.DoesNotExist:
+            raise AuthenticationFailed("User not found!")
+
+        serializer = DogWalkerSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(user=request.user)
+            return Response(serializer.data, status=201)
+        return Response(serializer.errors, status=400)
+
+    def put(self, request, pk):
```

Comment:
you dont need the pk here either, just get it from the payload

Reply:
all of these comments apply for the views below, so i dont pollute the pr with the same stuff